### PR TITLE
Refresh Dependencies and AI Documents

### PR DESCRIPTION
**Pull request by vmelamed**
# Summary

Refresh Dependencies and AI Documents

## Commits

- chore(diff-shared): update dependencies; copilot-instructions
- chore: rename
- chore: update AI documentation and improve comments for clarity

## Checklist

- [x] Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] Tests pass locally, and cover new code and edge cases. Test coverage is above 80%.
- [x] It is not expected that the performance will degrade by more than 20%
- [x] Documentation is updated if necessary

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -21,7 +21,7 @@
 *.xml text eol=lf
 
 # NuGet lockfiles are generated — never merge them by hand. The 'nugetlock' merge driver (defined locally by
-# repo-setup.sh as merge.nugetlock.*) takes the incoming side; regenerate with 'dotnet restore --force-evaluate'.
+# setup-repo.sh as merge.nugetlock.*) takes the incoming side; regenerate with 'dotnet restore --force-evaluate'.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .gitattributes]: it's OK

File: .github/copilot-instructions.md [hunk 2]
```diff
@@ -4,10 +4,18 @@
...
+- By default, the timestamp provider uses `System.TimeProvider.System.GetUtcNow().ToUnixTimeMilliseconds()` converted to Unix epoch time in milliseconds. If you need a different source of time, e.g. for testing purposes, you can use `Microsoft.Extensions.Time.Testing.FakeTimeProvider` (package `Microsoft.Extensions.TimeProvider.Testing`) or provide your own implementation that overrides the .NET BCL class `System.TimeProvider` and pass that to the factory.
+- The `vm2.Ulid` type is marked with the `System.Text.Json.Serialization.JsonConverterAttribute` attribute, so it can be serialized and deserialized by `System.Text.Json` without any additional  configuration. For Newtonsoft.Json, use the companion package `vm2.UlidSerialization.NsJson` ([source code](https://github.com/vmelamed/vm2.Ulid/blob/main/src/UlidNsConverter)).
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply .github/copilot-instructions.md]: fixed

File: CLAUDE.md [hunk 3]
```diff
@@ -14,43 +14,54 @@
...
-Key design decisions:
+### Key Design Decisions
 
-- `Ulid` is a `readonly record struct` (128-bit value, no heap allocation)
+- `Ulid` is a `readonly record struct` (128-bit value)
 - `UlidFactory` tracks timestamp and last random bytes for same-millisecond monotonicity
-- Randomness and timestamp are injected via `IRandomNumberGenerator` and
-  `ITimestampProvider` — testable, DI-friendly
+- Randomness and timestamp are injected via `IRandomNumberGenerator` and `System.TimeProvider` — testable, DI-friendly
 - `Ulid.NewUlid()` static method uses an internal singleton factory (convenience API)
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply CLAUDE.md]: fixed

File: CLAUDE.md [hunk 4]
```diff
@@ -14,43 +14,54 @@
...
+- ULID factories MUST be thread-safe and ensure monotonicity of generated ULIDs across application contexts. The factory uses two providers: one for the random bytes and one for the timestamp.
+- Use dependency injection to construct the factory and manage the providers. DI keeps the provider lifetimes explicit, makes testing simple, and enforces a single, consistent configuration across the app or service.
+- In simple scenarios, use the static method `vm2.Ulid.NewUlid()` instead of `vm2.UlidFactory`. It uses an internal single static factory instance with a cryptographic random number generator and a clock based on `System.TimeProvider.System.GetUtcNow().ToUnixTimeMilliseconds()`.
+- By default the `vm2.UlidFactory` uses a thread-safe, cryptographic random number generator (`vm2.UlidRandomProviders.CryptoRandom`). For a different source of randomness, use the  pseudo-random number generator `vm2.UlidRandomProviders.PseudoRandom` or provide your own implementation of `vm2.IRandomNumberGenerator` to the factory.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply CLAUDE.md]: fixed

File: CLAUDE.md [hunk 5]
```diff
@@ -63,14 +74,11 @@ vm2.Ulid/
 
 ## Active Work / Known Issues
 
-- [Fill in when working on this package]
+- Internally the Ulid is stored as a 128-bit value - `ReadOnlyMemory<byte>` (wrapping `byte[16]`) array - a reference type wrapper. Is this the most efficient way. Could a value based `ulong low; ulong high` or `Vector128<byte>` provide better performance or memory layout? Can the performance of the Ulid generation and conversions to/from string be improved by using a different internal representation?
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply CLAUDE.md]: better?

File: CLAUDE.md [hunk 6]
```diff
@@ -63,14 +74,11 @@ vm2.Ulid/
...
 
-- When working on `Ulid` struct internals: note it is a `readonly record struct`;
-  all operations must be non-mutating.
-- When working on `UlidFactory`: the monotonicity invariant is the core correctness
-  concern — any change must preserve it under concurrent access.
-- When writing tests: always inject `ITimestampProvider` and `IRandomNumberGenerator`
-  for determinism. Never rely on live clock or RNG in unit tests.
+- When working on `Ulid` struct internals: note it is a `readonly record struct`; all operations must be non-mutating.
+- When working on `UlidFactory`: the monotonicity invariant is the core correctness concern — any change must preserve it under concurrent access.
+- When writing tests: always inject `System.TimeProvider` and `IRandomNumberGenerator` for determinism. Never rely on live clock or RNG in unit tests.
```
[the remark this answers is not in the document]
Comment (vmelamed) [reply CLAUDE.md]: fixed
